### PR TITLE
migrates test_warnings.py from testdir to pytester

### PR DESCRIPTION
**Pull request by bot2x**

### CODE REVIEW

File: testing/test_warnings.py [hunk 1]
```diff
@@ -39,9 +39,9 @@ def foo():
 
 
 @pytest.mark.filterwarnings("default")
-def test_normal_flow(testdir, pyfile_with_warnings):
+def test_normal_flow(pytester: Pytester, pyfile_with_warnings):
```
Comment (bluetech) [inline testing/test_warnings.py:42]:
```suggestion
def test_normal_flow(pytester: Pytester, pyfile_with_warnings) -> None:
```

**Review by bluetech — COMMENTED** [review]
Thanks @bot2x, looks good to me!

One thing that would also be nice, while we're at it, is to add `-> None` to the test functions. I made one suggestion in the GitHub review but it also applies to the other places.

The reason for adding `-> None` is that without it mypy still considers the type annotations to be partial.

But it's also good without it, let me know if you can't get to it.

File: testing/test_warnings.py (hunk 1 above)
Comment (bot2x) [reply testing/test_warnings.py:42]: Understood. This is to be added to all the function which don't return anything for the completeness. I'll update the pr.

**Review by bluetech — APPROVED** [review]
Thanks @bot2x!

**Pull request merged**